### PR TITLE
2.20.0

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Channel: ❄️ 
Upstream: https://github.com/mlpack/ensmallen

Depends on:
* https://github.com/AnacondaRecipes/armadillo-feedstock/pull/1

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,14 +17,20 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - cmake
-    - make
+    - make  # [not win]
     - ninja  # [win]
+  host:
+    # ensmallen is a C++ header only library. We don't technically need these in hosts, but they
+    # allow ot inject run_exports.
+    - intel-openmp  # [win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - intel-openmp {{mkl}} # [blas_impl == "mkl"]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -17,14 +17,20 @@ requirements:
...
     - llvm-openmp  # [osx]
     - libgomp      # [linux]
-  host:
     - armadillo >=9.800.0
-    - libblas
   run:
+    # Pins through run_exports of packages in host.
+    - intel-openmp  # [win]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - {{ pin_compatible('intel-openmp') }} # [blas_impl == "mkl"]
```
intel-openmp doens't have run_exports.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -17,14 +17,20 @@ requirements:
...
     - {{ compiler('cxx') }}
     - cmake
-    - make
+    - make  # [not win]
     - ninja  # [win]
+  host:
+    # ensmallen is a C++ header only library. We don't technically need these in hosts, but they
+    # allow ot inject run_exports.
+    - intel-openmp  # [win]
     - llvm-openmp  # [osx]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -17,14 +17,20 @@ requirements:
...
     - libgomp      # [linux]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -17,14 +17,20 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - cmake
-    - make
+    - make  # [not win]
     - ninja  # [win]
```
Comment (cbouss) [inline recipe/meta.yaml:21]:
```suggestion
    - ninja  # [win]
    - libgomp # [linux and blas_impl != "mkl"]
    - llvm-openmp # [osx and blas_impl != "mkl"]
```
libgomp and llvm-openmp have strong run_exports, which means they have to be placed in build for it to be effective.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -17,14 +17,20 @@ requirements:
...
+    - intel-openmp  # [win]
     - llvm-openmp  # [osx]
     - libgomp      # [linux]
-  host:
     - armadillo >=9.800.0
-    - libblas
   run:
+    # Pins through run_exports of packages in host.
+    - intel-openmp  # [win]
+    - llvm-openmp  # [osx]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
To avoid more selector madness

File: recipe/meta.yaml [hunk 7]
```diff
@@ -17,14 +17,20 @@ requirements:
...
+    - libgomp      # [linux]
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
To avoid more selector madness

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: Thanks. In this case, we can just remove intel-openmp from host and do `intel-openmp {{ mkl }} # [blas_impl == "mkl"]` here.

**Review by cbouss — APPROVED** [review]

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Pull request merged**